go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -64,12 +61,12 @@ func (npw *nodePortWatcher) AddService(service *kapi.Service) {
 						"%d, stderr: %q, error: %v", npw.gwBridge, svcPort.NodePort, stderr, err)
 				}
 			}
+		}
+		if util.ServiceTypeHasClusterIP(service) {
```

Comment:
this is not needed if we recover the first if that discriminate the services we don't want

Reply:
done